src/System.Diagnostics.Process/src/System/Diagnostics/ProcessThread.OSX.cs

```diff
@@ -14,13 +16,22 @@ private ThreadPriorityLevel PriorityLevelCore
         {
             get
             {
-                // TODO: Implement this
-                throw NotImplemented.ByDesign;
+                int nice = 0;
+                int result = Interop.libc.getpriority(Interop.libc.PriorityWhich.PRIO_DARWIN_THREAD, _threadInfo._threadId, out nice);
+                if (result != 0)
+                {
+                    throw new System.ComponentModel.Win32Exception(Marshal.GetLastWin32Error());
```

Comment:
This is fine, but just FYI, the default ctor of Win32Exception does Marshal.GetLastWin32Error(), so you could simplify such calls to just ```throw new Win32Exception();```